quickstartKotlin/src/main/java/com/twilio/video/quickstart/kotlin/VideoActivity.kt

```diff
@@ -0,0 +1,669 @@
+        setDisconnectAction()
+    }
+
+    /*
+     * The initial state when there is no active room.
+     */
+    private fun initializeUI() {
+        connectActionFab.setImageDrawable(ContextCompat.getDrawable(this,
+                R.drawable.ic_video_call_white_24dp))
+        connectActionFab.show()
+        connectActionFab.setOnClickListener(connectActionClickListener())
+        switchCameraActionFab.show()
+        switchCameraActionFab.setOnClickListener(switchCameraClickListener())
+        localVideoActionFab.show()
+        localVideoActionFab.setOnClickListener(localVideoClickListener())
+        muteActionFab.show()
+        muteActionFab.setOnClickListener(muteClickListener())
+    }
+
+    /*
+     * The actions performed during disconnect.
+     */
+    private fun setDisconnectAction() {
+        connectActionFab.setImageDrawable(ContextCompat.getDrawable(this,
+                R.drawable.ic_call_end_white_24px))
+        connectActionFab.show()
+        connectActionFab.setOnClickListener(disconnectClickListener())
+    }
+
+    /*
+     * Creates an connect UI dialog
+     */
+    private fun showConnectDialog() {
+        val roomEditText = EditText(this)
+        alertDialog = createConnectDialog(roomEditText,
+                connectClickListener(roomEditText), cancelConnectDialogClickListener(), this)
+        alertDialog!!.show()
+    }
+
+    /*
+     * Called when participant joins the room
+     */
+    private fun addParticipant(participant: Participant) {
+        /*
+         * This app only displays video for one additional participant per Room
+         */
+        if (thumbnailVideoView.visibility == View.VISIBLE) {
+            Snackbar.make(connectActionFab,
+                    "Multiple participants are not currently support in this UI",
+                    Snackbar.LENGTH_LONG)
+                    .setAction("Action", null).show()
+            return
+        }
+        participantIdentity = participant.identity
+        videoStatusTextView.text = "Participant $participantIdentity joined"
+
+        /*
+         * Add participant renderer
+         */
+        if (participant.videoTracks.isNotEmpty()) {
+            addParticipantVideo(participant.videoTracks.first())
+        }
+
+        /*
+         * Start listening for participant events
+         */
+        participant.setListener(participantListener)
+    }
+
+    /*
+     * Set primary view as renderer for participant video track
+     */
+    private fun addParticipantVideo(videoTrack: VideoTrack) {
+        moveLocalVideoToThumbnailView()
+        primaryVideoView.mirror = false
+        videoTrack.addRenderer(primaryVideoView)
+    }
+
+    private fun moveLocalVideoToThumbnailView() {
+        if (thumbnailVideoView.visibility == View.GONE) {
+            thumbnailVideoView.visibility = View.VISIBLE
+            localVideoTrack?.removeRenderer(primaryVideoView)
```

Comment:
Multiple method invocations can be called using function [with](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/with.html).